nutanix/resource_nutanix_virtual_machine_clone.go

```diff
@@ -0,0 +1,787 @@
+	if ero != nil {
+		log.Printf("[WARN] could not get the IP for VM(%s): %s", cloneVMUUID, err)
+	} else {
+		vm := vmIntentResponse.(*v3.VMIntentResponse)
+
+		if len(vm.Status.Resources.NicList) > 0 && len(vm.Status.Resources.NicList[0].IPEndpointList) != 0 {
+			d.SetConnInfo(map[string]string{
+				"type": "ssh",
+				"host": *vm.Status.Resources.NicList[0].IPEndpointList[0].IP,
+			})
+		}
+	}
+
+	d.SetId(cloneVMUUID)
+	//return nil
+	return resourceNutanixVirtualMachineCloneRead(ctx, d, meta)
+}
+
+func resourceNutanixVirtualMachineCloneUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return nil
+}
+
+func resourceNutanixVirtualMachineCloneRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return resourceNutanixVirtualMachineRead(ctx, d, meta)
+}
+
+func resourceNutanixVirtualMachineCloneDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return nil
+}
+
+func getMetadataCloneAttributes(d *schema.ResourceData) (out *v3.Metadata) {
+	resourceData, ok := d.GetOk("metadata")
+	if !ok {
+		return nil
+	}
+
+	meta := resourceData.([]interface{})[0].(map[string]interface{})
+
+	if name, ok := meta["uuid"]; ok {
+		out.UUID = utils.StringPtr(name.(string))
+	}
+
+	if name, ok := meta["entity_version"]; ok {
+		out.EntityVersion = utils.StringPtr(name.(string))
+	}
+	return out
+}
+
+func expandOverrideSpec(d *schema.ResourceData) *v3.OverrideSpec {
+	res := &v3.OverrideSpec{}
+
+	if name, ok := d.GetOk("name"); ok {
+		res.Name = utils.StringPtr(name.(string))
+	}
+
+	if numSockets, sok := d.GetOk("num_sockets"); sok {
+		res.NumSockets = utils.IntPtr(numSockets.(int))
+	}
+
+	if vcpuSock, vok := d.GetOk("num_vcpus_per_socket"); vok {
+		res.NumVcpusPerSocket = utils.IntPtr(vcpuSock.(int))
+	}
+
+	if numThreads, ok := d.GetOk("num_threads_per_core"); ok {
+		res.NumThreadsPerCore = utils.IntPtr(numThreads.(int))
+	}
+
+	if memorySize, mok := d.GetOk("memory_size_mib"); mok {
+		res.MemorySizeMib = utils.IntPtr(memorySize.(int))
+	}
+
+	if _, nok := d.GetOk("nic_list"); nok {
+		res.NicList = expandNicList(d)
+	}
+
+	guestCustom := &v3.GuestCustomization{}
+	cloudInit := &v3.GuestCustomizationCloudInit{}
+
+	if v, ok := d.GetOk("guest_customization_cloud_init_user_data"); ok {
+		cloudInit.UserData = utils.StringPtr(v.(string))
+	}
+
+	if v, ok := d.GetOk("guest_customization_cloud_init_meta_data"); ok {
+		cloudInit.MetaData = utils.StringPtr(v.(string))
+	}
+
+	if v, ok := d.GetOk("guest_customization_cloud_init_custom_key_values"); ok {
+		cloudInit.CustomKeyValues = utils.ConvertMapString(v.(map[string]interface{}))
+	}
+
+	if !reflect.DeepEqual(*cloudInit, (v3.GuestCustomizationCloudInit{})) {
+		guestCustom.CloudInit = cloudInit
+	}
+
+	if v, ok := d.GetOk("guest_customization_is_overridable"); ok {
+		guestCustom.IsOverridable = utils.BoolPtr(v.(bool))
+	}
+	if v, ok := d.GetOk("guest_customization_sysprep"); ok {
```

Comment:
Most of it is duplicate code. We can reuse existing funcs.

Reply:
Reusing existing funcs will be cause spec to add more attributes which aren't required by clone payload.

Reply:
Reusing existing funcs will be cause spec to add more attributes which aren't required by clone payload.